src/main.cpp

```diff
@@ -3923,7 +3923,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     }
 
 
-    else if (strCommand == "getaddr")
+    else if ((strCommand == "getaddr") && (pfrom->fInbound))
```

Comment:
Thinking of it, this really needs a comment in the source code why this asymmetric behavior was introduced. Someone reading the code is bound to wonder why.